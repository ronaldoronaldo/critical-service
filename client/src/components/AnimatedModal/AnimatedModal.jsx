import React, {useRef} from "react";
import {Background, Container, Modal} from "./AnimatedModal.style";
import useClickOutside from "../../hooks/useClickOutside";

export default function AnimatedModal({isOpen, closeModal, children}) {
  const wrapperRef = useRef(null);
  useClickOutside(wrapperRef, closeModal);

  return (
    <Container isOpen={isOpen}>
      <Background isOpen={isOpen}>
        <Modal ref={wrapperRef} isOpen={isOpen}>
          {children}
        </Modal>
      </Background>
    </Container>
  );
}
