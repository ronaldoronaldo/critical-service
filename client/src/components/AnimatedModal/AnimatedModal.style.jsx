import styled from 'styled-components'
import {colors, radius} from "../../styles/styleGuide";

export const Container = styled.div`
  ${({ isOpen }) => `
    position: fixed;
    width: 100%;
    height: 100%;
    transition-delay: ${isOpen ? '0' : '0.8s'};
    z-index: ${isOpen ? 5 : -1};
    top: 0;
  `}
`

export const Background = styled.div`
  ${({ isOpen }) => `
    width: 100%;
    height: 100%;
    background-color: ${colors.modalBackground};
    opacity: ${isOpen ? 1 : 0};
    transition: opacity 0.4s ease-in-out;
    display: flex;
    justify-content: center;
    align-items: center;
  `}
`

export const Modal = styled.div`
  ${({ isOpen }) => `
    background-color: ${colors.white};
    border-radius: ${radius};
    position: relative;
    transition: top 0.4s ease-in-out;
    top: ${isOpen ? '0' : '-100vh'};
  `}
`
