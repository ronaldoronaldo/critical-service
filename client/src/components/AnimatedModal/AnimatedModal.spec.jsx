import React from "react";
import { render } from "@testing-library/react";

import AnimatedModal from "./AnimatedModal";

describe("<AnimatedModal />", () => {
  it("should match snapshot if is open", () => {
    const { container } = render(<AnimatedModal isOpen />);
    expect(container).toMatchSnapshot();
  });

  it("should match snapshot if is closed", () => {
    const { container } = render(<AnimatedModal isOpen={false} />);
    expect(container).toMatchSnapshot();
  });
});
