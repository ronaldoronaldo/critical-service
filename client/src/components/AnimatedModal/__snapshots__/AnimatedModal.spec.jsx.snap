// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AnimatedModal /> should match snapshot if is closed 1`] = `
<div>
  <div
    class="sc-AxjAm horUeu"
  >
    <div
      class="sc-AxirZ bbGxNM"
    >
      <div
        class="sc-AxiKw dLqinz"
      />
    </div>
  </div>
</div>
`;

exports[`<AnimatedModal /> should match snapshot if is open 1`] = `
<div>
  <div
    class="sc-AxjAm czIcey"
  >
    <div
      class="sc-AxirZ WDNph"
    >
      <div
        class="sc-AxiKw fVhjZd"
      />
    </div>
  </div>
</div>
`;
